lib/adapters/cryptopus_adapter.rb

```diff
@@ -52,8 +52,20 @@ def find_account_by_name(name)
     secret_account
   end
 
+  def renewed_auth_token
+    json = get("api_users/#{current_user_id}/token")
+    JSON.parse(json)['token']
+  end
+
   private
 
+  def current_user_id
+    users = JSON.parse(get('api_users'), symbolize_names: true)
+    users[:data].find do |user|
+      user[:attributes][:username] == session_adapter.session_data[:username]
+    end[:id]
```

Comment:
user can always be found, can it not?

Reply:
On a 200 it should always find the user. Would be veeeeery odd if not.